ionic-team/ionic-framework
# feat: add high contrast implementation

Issue number: Internal

---------





## What is the current behavior?


Users do not have a way of increasing the contrast in Ionic apps. This is valuable for people with low vision as increasing the contrast between foreground and background content helps improve readability.

## What is the new behavior?


- Adds a high contrast light and high contrast dark theme. As with our other themes, developers can choose between system, class, and always stylesheets.

While we aim to improve contrast for text and UI components, this feature prioritizes text in the event that both text and UI component cannot be improved without one negatively impacting the other.

## Does this introduce a breaking change?





## Other information



Dev build: `7.6.7-dev.11706894781.1cd59fde`

Testing instructions:

1. Open `src/themes/test/css-variables`. Activate the high contrast light and dark themes to verify that contrast does increase.
2. Use the dev build to integrate the theme into a test app (conference app, starter app, etc).

I'd recommend using these imports:

```css
@import "@ionic/angular/css/themes/high-contrast.system.css";
@import "@ionic/angular/css/themes/high-contrast-dark.system.css";
```
Note: Make sure this is imported **after** `core.scss`
